clouddrift/analysis.py

```diff
@@ -91,7 +92,9 @@ def apply_ragged(
 
 
 def segment(
-    x: np.ndarray, tolerance: float, rowsize: np.ndarray[int] = None
+    x: np.ndarray,
+    tolerance: float | timedelta,
```

Comment:
This syntax was introduced in Python 3.10. If we want to support Python 3.9, use Union:

```suggestion
    tolerance: Union[float, timedelta],
```

(but I personally am fine dropping 3.9 for improved type hint syntax)

Reply:
I will make the tests run with 3.9 and 3.10 for now. So we can catch this type of error.. because I'm only using 3.10 (or 3.11) currently.